### PR TITLE
Create backend plugin through CLI

### LINKED ISSUE #963: create-plugin --backend
## Feature Suggestion

We should have a backend plugin version of `backstage-cli create-plugin`.

## Possible Implementation

We'd need a template for backend plugins in `packages/cli/templates`, an a way to specify that you want to create a backend plugin. It could for example be done via a `--backend` or `--template=backend` flag. Could also be a separate command or an interactive choice during the cli wizard.

## Context

We want backend plugin creation to be as smooth as frontend plugin creation :grin:

### PR DESCRIPTION
**Pull request by benjdlambert**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by Rugvip** [comment]
@benjdlambert /poke

**Comment by benjdlambert** [comment]
Wonder if we could move this to Hacktober fest? @Rugvip

**Comment by Rugvip** [comment]
We could, it's now very related to https://github.com/spotify/backstage/issues/2491 too

File: packages/cli/templates/default-backend-plugin/package.json.hbs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@backstage/plugin-{{id}}-backend",
+  "version": "{{version}}",
+  "main": "src/index.ts",
+  "types": "src/index.ts",
+  "license": "Apache-2.0",
+  "private": true,
+  "publishConfig": {
+    "access": "public",
+    "main": "dist/index.esm.js",
```
Comment (Rugvip) [inline packages/cli/templates/default-backend-plugin/package.json.hbs:10]: should be `cjs`

File: packages/cli/src/commands/create-plugin/createPlugin.ts [hunk 2]
```diff
@@ -175,7 +175,8 @@ export async function movePlugin(
   });
 }
 
-export default async () => {
+export default async ({ backend }: { backend: boolean }) => {
+  console.warn(backend);
```
Comment (Rugvip) [inline packages/cli/src/commands/create-plugin/createPlugin.ts:179]: No

File: packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
 import { ThemeProvider } from '@material-ui/core';
 import { lightTheme } from '@backstage/theme';
 
 describe('ExampleComponent', () => {
   it('should render', () => {
-    mockFetch.mockResponse(() => new Promise(() => {}));
     const rendered = render(
       <ThemeProvider theme={lightTheme}>
-        <ExampleComponent />
+      <ExampleComponent />
```
Comment (Rugvip) [inline packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs:11]: Indentation looks off here

**Review by Rugvip — COMMENTED** [review]
Would be nice to sync this up with the changes done to the plugin template in #2316

msw 🎉

**Comment by benjdlambert** [comment]
Some of this stuff drifted a little so will repatch and update. 👍

File: plugins/circleci/package.json [hunk 4]
```diff
@@ -46,7 +46,9 @@
     "@types/jest": "^26.0.7",
     "@types/node": "^12.0.0",
     "@types/react-lazylog": "^4.5.0",
-    "jest-fetch-mock": "^3.0.3"
+    "jest-fetch-mock": "^3.0.3",
+    "msw": "^0.20.5",
```
Comment (soapraj) [inline plugins/circleci/package.json:50]: these deps don't make sense for frontend plugins right?

**Review by soapraj — APPROVED** [review]

File: plugins/circleci/package.json (hunk 4 above)
Comment (soapraj) [reply plugins/circleci/package.json:50]: @benjdlambert I think this fix is to stop the plugin: diff command to stop complaining but I think it needs to be changed to compare frontend and backend plugin separately

**Review by hooloovooo — APPROVED** [review]

**Review by emmaindal — APPROVED** [review]

File: plugins/circleci/package.json (hunk 4 above)
Comment (benjdlambert) [reply plugins/circleci/package.json:50]: I want to standardise that anyway. `jest-fetch-mock` will disappear in favour of `node-fetch` and `msw` for both backend and frontend plugins. All the tests run in `node` so it doesn't matter either way.

**Pull request merged**